Rename position-area: x/y-self-start/end to self-x/y-start/end

fantasai opened the pull request:
#### 651ff0c182e282bb5b278816b322560cf882a38d
<pre>
Rename position-area: x/y-self-start/end to self-x/y-start/end
<a href="https://bugs.webkit.org/show_bug.cgi?id=300399">https://bugs.webkit.org/show_bug.cgi?id=300399</a>
<a href="https://rdar.apple.com/162214793">rdar://162214793</a>

Reviewed by Tim Nguyen.

CSSWG renamed these keywords after multiple people were confused.
<a href="https://github.com/w3c/csswg-drafts/issues/11371">https://github.com/w3c/csswg-drafts/issues/11371</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-align-justify-wm-dir-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-align-justify-wm-dir.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-in-position-try-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-in-position-try.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-wm-dir-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-wm-dir.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/try-tactic-position-area-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/try-tactic-position-area.html:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Anchor.cpp:
(WebCore::CSSPropertyParserHelpers::getKeywordType):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::positionAreaKeywordToAxis):
(WebCore::Style::positionAreaKeywordToTrack):
(WebCore::Style::positionAreaKeywordToSelf):
* Source/WebCore/style/StyleExtractorConverter.h:
(WebCore::Style::ExtractorConverter::convertPositionArea):

Canonical link: <a href="https://commits.webkit.org/301226@main">https://commits.webkit.org/301226@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8854daf2158e8c9a10e707cec52507415b09856a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125284 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44949 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132139 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77155 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9f59cdae-ae3f-4cae-a42c-a9c7b223d2c7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127161 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45637 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53516 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95389 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63324 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/59332d92-0df2-469b-a93c-026d155bc54c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128238 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36444 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112036 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75930 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/913607fc-0029-4a1e-b523-da79d9a883a0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35343 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30204 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75610 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106211 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30429 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134818 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52088 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39872 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103857 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52526 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108257 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103620 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26395 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48981 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27269 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49194 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51983 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57764 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51344 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54697 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53035 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

tuankiet65 reviewed (commented):
Looks good to me, but I'm not a reviewer ¯⁠\\_⁠(⁠ツ⁠)⁠\_⁠/⁠¯

fantasai committed:
Rename position-area: x/y-self-start/end to self-x/y-start/end
https://bugs.webkit.org/show_bug.cgi?id=300399
rdar://162214793

Reviewed by Tim Nguyen.

CSSWG renamed these keywords after multiple people were confused.
https://github.com/w3c/csswg-drafts/issues/11371

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-align-justify-wm-dir-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-align-justify-wm-dir.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-in-position-try-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-in-position-try.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-wm-dir-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-wm-dir.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/try-tactic-position-area-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/try-tactic-position-area.html:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Anchor.cpp:
(WebCore::CSSPropertyParserHelpers::getKeywordType):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::positionAreaKeywordToAxis):
(WebCore::Style::positionAreaKeywordToTrack):
(WebCore::Style::positionAreaKeywordToSelf):
* Source/WebCore/style/StyleExtractorConverter.h:
(WebCore::Style::ExtractorConverter::convertPositionArea):

Canonical link: https://commits.webkit.org/301226@main
